File: frontend/apps/quarry/src/types/FrameworkDetails.ts
import type { FrameworkSummary } from "./FrameworkSummary";
import type { FrameworkCapability } from "./FrameworkCapability";
import type { FrameworkComponentDescriptor } from "./FrameworkComponentDescriptor";

export type FrameworkDetails = {
  summary: FrameworkSummary;
  capabilities: FrameworkCapability[];
  useCases: string[];
  components: FrameworkComponentDescriptor[];
};

File: backend/src/Quarry.Application/Catalog/FrameworkDetails.cs
namespace Quarry.Application.Catalog;

public sealed record FrameworkDetails(
    FrameworkSummary Summary,
    IReadOnlyList<FrameworkCapability> Capabilities,
    IReadOnlyList<string> UseCases,
    IReadOnlyList<FrameworkComponentDescriptor> Components, PreviewManifest? PreviewManifest = null);
